src/librustc_target/abi/mod.rs

```diff
@@ -784,6 +784,14 @@ impl Abi {
             _ => false,
         }
     }
+
+    /// Returns true if this is an uninhabited type
+    pub fn is_uninhabited(&self) -> bool {
```

Comment:
For some reason I would've avoided this but I guess it's fine.
Could also go through `rg '==.*::Uninhabited'` and change all of those to use `is_uninhabited`?